### ISSUE TITLE
Gtk Error: Gtk 2 and 3 mix

### ISSUE THREAD

**Issue by JoeKays** [xournalpp/xournalpp#292]
When I patch the cmath issue and build (sucessfully), starting the executable results in:

> Gtk-ERROR **: GTK+ 2.x symbols detected. Using GTK+ 2.x and GTK+ 3 in the same process is not supported

**Comment by JoeKays** [comment]
I did a clean build after pulling from master and only changed the code to fix the cmath issue.
So yeah, really strange.

**Comment by JoeKays** [comment]
And I compiled with **CMAKE_CXX_FLAGS="-fpermissive"** if this has anything to do with it. Without the flag it does not compile.

**Comment by andreasb242** [comment]
I merged the GTK3 branch to master, and did a lot of additional fixes (come caused by the GTK3 merge).
Over 8'000 Lines changed, because of this I wrote this to the Readme ;-)

**Can you tell which distribution and which version do you use, so I may can reproduce your problem?**

**Comment by Febbe** [comment]
What is your compiler, including the version?
Which is the stdlib you are using?
What is the operating system?
How you build xournal++; including the commando you used?

And yes it could've something to do with the fpermissive flag. This means that the compiler will compile the software even if the code does not follow the language rules. 
What is the error without it?

**Comment by JoeKays** [comment]
The -fpermissive flag ist required to solve the following issue:
```
src/pdf/popplerdirect/PdfObjectWriter.cpp:88:35: error: invalid conversion from ‘const GooString*’ to ‘GooString*’ [-fpermissive]
   this->writeString(obj->getString());
                     ~~~~~~~~~~~~~~^~
```
```
src/pdf/popplerdirect/PdfObjectWriter.h:28:30: note:   initializing argument 1 of ‘void PdfObjectWriter::writeString(GooString*)’
  void writeString(GooString* s);
                   ~~~~~~~~~~~^
```

**Comment by JoeKays** [comment]
Update: Merge #310 fixed the -fpermissive issue but build without it also results in the same GTK error.
Not sure though what this has to do with the building as it is clearly GTK related.
What info do you need?

**Comment by JoeKays** [comment]
CMakeCache.txt lines with 'gtk':

[CMakeCache_gtk.txt](https://github.com/xournalpp/xournalpp/files/2592776/CMakeCache_gtk.txt)

What does that mean?

**Comment by andreasb242** [comment]
The file looks similar to mine, there are some differences, but I'm not a CMake specialist, I don't see an error.

Can you tell us which distribution (and version) you are using?
Somebody needs to reproduce your problem, to find it. Without this information it will not be possible.

**Comment by JoeKays** [comment]
I am on debian buster.

**Comment by andreasb242** [comment]
I'm currently downloading Debian Buster, and then I'll set up a Notebook and do a short test.
If I have enough Time I may do this today, but not sure.

**Comment by taaem** [comment]
This also happens on Fedora 29

**Comment by andreasb242** [comment]
So there is something wrong with our CMake file. If you can help, I'm open for any tips...

I'm currently installing debian, to try to reproduce the problem.

**Comment by andreasb242** [comment]
I installed Debian stretch (by accident, realized to late).
With this version the error does not occur. It is working...

**Comment by andreasb242** [comment]
I have now Debian buster installed, compiling and all is working as excepted. Can you please also copy the cmake command, did you add somting, or did you use "cmake .." in the build directory?

Maybe git clone the repo on a new location and try again?

**Comment by taaem** [comment]
I tried with Mathtex support on and off.
Do you have gtk2-dev stuff installed? Maybe it only picks that up if it is installed

**Comment by andreasb242** [comment]
No, this is a complete fresh install, I only installed git and the dependencies for Xournal++ (which are in the Readme of this repo). Nothing else.

So this is possible.

**Comment by taaem** [comment]
Okay, maybe you could try installing that?
Btw I'll try to give you a complete list of the fedora dependencies.

**Comment by andreasb242** [comment]
Confirmed: You are right! After installing libgtk2.0-dev I got the same error.
(But it's not fixed, I'm not the CMake / Build specialist. But hopefully I find the reason.)

If you don't need GTK2, uninstalling might fix it. If you have application which need the GTK2-dev package, this is not a solution...

**Comment by JoeKays** [comment]
Can confirm as well!
For a quick and dirty solution it works when I remove the *values* (you have to keep the keys!) of all the keys starting with 'GTK2' from CMakeCache.txt.
Then it compiles beautifully as always and also then the error when running the executable is gone, too!

**Comment by siliconninja** [comment]
> Can confirm as well!
> For a quick and dirty solution it works when I remove the _values_ (you have to keep the keys!) of all the keys starting with 'GTK2' from CMakeCache.txt.
> Then it compiles beautifully as always and also then the error when running the executable is gone, too!

Here's a diff with my changes to the CMakeCache.txt and my full CMakeCache.txt file with the fix applied (might vary if you're on a different distro, I'm on Manjaro.) **Note that you have to generate the CMakeCache.txt file first using `cmake ..` in the `build` directory you created.**

**Another note for the CMakeCache-fix.txt file: directory names (like /home/siliconninja/Documents/Linux Compiled Programs/xournalpp/build), etc and library file locations like /usr/lib/libcairo.so need to be changed depending on distro to get it working.**

(diff -u won't actually show the changes to each line, so I used diff -N and diff: both contents were the same for reference).
Diff: [CMakeCache-GTK2RemoveValues-Workaround.patch.txt](https://github.com/xournalpp/xournalpp/files/2602682/CMakeCache-GTK2RemoveValues-Workaround.patch.txt)
And the entire file with the fix applied to it for CMakeCache.txt: [CMakeCache-fix.txt](https://github.com/xournalpp/xournalpp/files/2602649/CMakeCache-fix.txt))

**Comment by andreasb242** [comment]
But why?
There are no GTK2 dependencies?!

**Comment by andreasb242** [comment]
I have an Idea, and maybe somebody has time.

Checkout a copy of Xournalpp to a new folder, delete all source files.
Create one single file, e.g. use this example:
https://developer.gnome.org/gtk3/stable/gtk-getting-started.html

Is it working? Is it also not working.
If it is not working, strip dependencies down, until we know the problem.

Should be relative easy, but needs much time...

**Comment by siliconninja** [comment]
> Create one single file, e.g. use this example:
> https://developer.gnome.org/gtk3/stable/gtk-getting-started.html
> 
> Is it working? Is it also not working.


I was able to get the example-0.c file to compile by using the command on their website: ````gcc `pkg-config --cflags gtk+-3.0 -o example-0 example-0.c `pkg-config --libs gtk+-3.0` ````. No dependency issues there.

I also tried messing around with the CMakeLists.txt file in the src/ directory by removing all dependencies on the src files and just getting example-0.c to compile when doing mkdir build, cmake .. and make. I encounter an error when running make where I get multiple main() statements in example-0.c, maybe something's wrong with the way I'm using CMake for that particular C file.

I tried enabling DEV_CHECK_GTK3_COMPAT in CMakeLists.txt on the original and it wouldn't compile.

I tried using many tools to resolve the dependencies to see what could be cross linking with GTK2 and GTK3. I didn't find anything conflicting with GTK2 and GTK3 packages. I even used pkg-config, ldd, and lddtree by ncopa (https://github.com/ncopa/lddtree). I didn't find any packages that conflict with gtk+-2.0 and gtk+-3.0: I tried adding more libraries in the CMakeLists.txt file but that didn't help.

I referred to:
https://stackoverflow.com/questions/20714343/gtk-error-gtk-2-x-symbols-detected
https://stackoverflow.com/questions/42534129/opencv-error-gtk-2-x-symbols-detected-using-gtk-2-x-and-gtk-3-in-the-same-p
(and many other links, those 2 were the most helpful IMO)

There were no other binaries that were made (I tried .o files but those are not executables, and ldd didn't like them) so I had to analyze just the xournalpp binary.


<details> 
<summary>pkg-config output</summary>

````
$ pkg-config --libs gtk+-2.0 glib-2.0 gthread-2.0 libxml-2.0
-lgtk-x11-2.0 -lgdk-x11-2.0 -lpangocairo-1.0 -latk-1.0 -lcairo -lgdk_pixbuf-2.0 -lgio-2.0 -lpangoft2-1.0 -lpango-1.0 -lgobject-2.0 -lfontconfig -lfreetype -lgthread-2.0 -pthread -lglib-2.0 -lxml2 
$ pkg-config --libs glib-2.0 gthread-2.0 libxml-2.0
-lgthread-2.0 -pthread -lglib-2.0 -lxml2 
$ pkg-config --libs gtk+-3.0 glib-2.0 gthread-2.0 libxml-2.0
-lgtk-3 -lgdk-3 -lpangocairo-1.0 -lpango-1.0 -latk-1.0 -lcairo-gobject -lcairo -lgdk_pixbuf-2.0 -lgio-2.0 -lgobject-2.0 -lgthread-2.0 -pthread -lglib-2.0 -lxml2 
````

</details> 

<details> 
<summary>ldd output</summary>

````
$ ldd xournalpp 
        linux-vdso.so.1 (0x00007ffed81fb000)
        libboost_system.so.1.68.0 => /usr/lib/libboost_system.so.1.68.0 (0x00007fb469ecb000)
        libboost_filesystem.so.1.68.0 => /usr/lib/libboost_filesystem.so.1.68.0 (0x00007fb469ead000)
        libboost_locale.so.1.68.0 => /usr/lib/libboost_locale.so.1.68.0 (0x00007fb469dae000)
        libboost_iostreams.so.1.68.0 => /usr/lib/libboost_iostreams.so.1.68.0 (0x00007fb469d8d000)
        libboost_thread.so.1.68.0 => /usr/lib/libboost_thread.so.1.68.0 (0x00007fb469d60000)
        libboost_regex.so.1.68.0 => /usr/lib/libboost_regex.so.1.68.0 (0x00007fb469c4c000)
        libboost_chrono.so.1.68.0 => /usr/lib/libboost_chrono.so.1.68.0 (0x00007fb469c3f000)
        libboost_date_time.so.1.68.0 => /usr/lib/libboost_date_time.so.1.68.0 (0x00007fb469c2b000)
        libboost_atomic.so.1.68.0 => /usr/lib/libboost_atomic.so.1.68.0 (0x00007fb469c26000)
        libgtk-3.so.0 => /usr/lib/libgtk-3.so.0 (0x00007fb4694dd000)
        libgdk-3.so.0 => /usr/lib/libgdk-3.so.0 (0x00007fb4693e3000)
        libpangocairo-1.0.so.0 => /usr/lib/libpangocairo-1.0.so.0 (0x00007fb4693d3000)
        libpango-1.0.so.0 => /usr/lib/libpango-1.0.so.0 (0x00007fb469388000)
        libatk-1.0.so.0 => /usr/lib/libatk-1.0.so.0 (0x00007fb46935f000)
        libcairo-gobject.so.2 => /usr/lib/libcairo-gobject.so.2 (0x00007fb469353000)
        libcairo.so.2 => /usr/lib/libcairo.so.2 (0x00007fb46922c000)
        libgdk_pixbuf-2.0.so.0 => /usr/lib/libgdk_pixbuf-2.0.so.0 (0x00007fb469205000)
        libgio-2.0.so.0 => /usr/lib/libgio-2.0.so.0 (0x00007fb469063000)
        libgobject-2.0.so.0 => /usr/lib/libgobject-2.0.so.0 (0x00007fb46900b000)
        libglib-2.0.so.0 => /usr/lib/libglib-2.0.so.0 (0x00007fb468ee8000)
        libgthread-2.0.so.0 => /usr/lib/libgthread-2.0.so.0 (0x00007fb468ee3000)
        libxml2.so.2 => /usr/lib/libxml2.so.2 (0x00007fb468d7b000)
        libgmodule-2.0.so.0 => /usr/lib/libgmodule-2.0.so.0 (0x00007fb468d75000)
        libpangoft2-1.0.so.0 => /usr/lib/libpangoft2-1.0.so.0 (0x00007fb468d5f000)
        libpangoxft-1.0.so.0 => /usr/lib/libpangoxft-1.0.so.0 (0x00007fb468d53000)
        libgdk-x11-2.0.so.0 => /usr/lib/libgdk-x11-2.0.so.0 (0x00007fb468a9f000)
        libgtk-x11-2.0.so.0 => /usr/lib/libgtk-x11-2.0.so.0 (0x00007fb468462000)
        libpoppler.so.78 => /usr/lib/libpoppler.so.78 (0x00007fb4681b8000)
        libpoppler-glib.so.8 => /usr/lib/libpoppler-glib.so.8 (0x00007fb468158000)
        libz.so.1 => /usr/lib/libz.so.1 (0x00007fb467f41000)
        libpthread.so.0 => /usr/lib/libpthread.so.0 (0x00007fb467f1e000)
        libstdc++.so.6 => /usr/lib/libstdc++.so.6 (0x00007fb467d8f000)
        libm.so.6 => /usr/lib/libm.so.6 (0x00007fb467c0a000)
        libgcc_s.so.1 => /usr/lib/libgcc_s.so.1 (0x00007fb467bf0000)
        libc.so.6 => /usr/lib/libc.so.6 (0x00007fb467a2c000)
        librt.so.1 => /usr/lib/librt.so.1 (0x00007fb467a22000)
        libicudata.so.63 => /usr/lib/libicudata.so.63 (0x00007fb466032000)
        libicui18n.so.63 => /usr/lib/libicui18n.so.63 (0x00007fb465d54000)
        libicuuc.so.63 => /usr/lib/libicuuc.so.63 (0x00007fb465b84000)
        libbz2.so.1.0 => /usr/lib/libbz2.so.1.0 (0x00007fb465b71000)
        liblzma.so.5 => /usr/lib/liblzma.so.5 (0x00007fb46594b000)
        libX11.so.6 => /usr/lib/libX11.so.6 (0x00007fb46580a000)
        libXi.so.6 => /usr/lib/libXi.so.6 (0x00007fb4655f9000)
        libXfixes.so.3 => /usr/lib/libXfixes.so.3 (0x00007fb4653f3000)
        libatk-bridge-2.0.so.0 => /usr/lib/libatk-bridge-2.0.so.0 (0x00007fb4653bf000)
        libepoxy.so.0 => /usr/lib/libepoxy.so.0 (0x00007fb465290000)
        libharfbuzz.so.0 => /usr/lib/libharfbuzz.so.0 (0x00007fb4651ad000)
        libfontconfig.so.1 => /usr/lib/libfontconfig.so.1 (0x00007fb465164000)
        libfreetype.so.6 => /usr/lib/libfreetype.so.6 (0x00007fb464e9b000)
        libXinerama.so.1 => /usr/lib/libXinerama.so.1 (0x00007fb464c98000)
        libXrandr.so.2 => /usr/lib/libXrandr.so.2 (0x00007fb464a8d000)
        libXcursor.so.1 => /usr/lib/libXcursor.so.1 (0x00007fb464883000)
        libXcomposite.so.1 => /usr/lib/libXcomposite.so.1 (0x00007fb464680000)
        libXdamage.so.1 => /usr/lib/libXdamage.so.1 (0x00007fb46447b000)
        libxkbcommon.so.0 => /usr/lib/libxkbcommon.so.0 (0x00007fb46423c000)
        libwayland-cursor.so.0 => /usr/lib/libwayland-cursor.so.0 (0x00007fb464233000)
        libwayland-egl.so.1 => /usr/lib/libwayland-egl.so.1 (0x00007fb46422e000)
        libwayland-client.so.0 => /usr/lib/libwayland-client.so.0 (0x00007fb46421d000)
        libXext.so.6 => /usr/lib/libXext.so.6 (0x00007fb46400b000)
        libthai.so.0 => /usr/lib/libthai.so.0 (0x00007fb463ffe000)
        libfribidi.so.0 => /usr/lib/libfribidi.so.0 (0x00007fb463de2000)
        libpixman-1.so.0 => /usr/lib/libpixman-1.so.0 (0x00007fb463b3a000)
        libpng16.so.16 => /usr/lib/libpng16.so.16 (0x00007fb463b02000)
        libxcb-shm.so.0 => /usr/lib/libxcb-shm.so.0 (0x00007fb463afd000)
        libxcb.so.1 => /usr/lib/libxcb.so.1 (0x00007fb463ad3000)
        libxcb-render.so.0 => /usr/lib/libxcb-render.so.0 (0x00007fb463ac1000)
        libXrender.so.1 => /usr/lib/libXrender.so.1 (0x00007fb4638b6000)
        libmount.so.1 => /usr/lib/libmount.so.1 (0x00007fb463858000)
        libresolv.so.2 => /usr/lib/libresolv.so.2 (0x00007fb46383f000)
        libffi.so.6 => /usr/lib/libffi.so.6 (0x00007fb463834000)
        libpcre.so.1 => /usr/lib/libpcre.so.1 (0x00007fb4635c0000)
        libdl.so.2 => /usr/lib/libdl.so.2 (0x00007fb4635bb000)
        libXft.so.2 => /usr/lib/libXft.so.2 (0x00007fb4633a5000)
        libjpeg.so.8 => /usr/lib/libjpeg.so.8 (0x00007fb463310000)
        libcurl.so.4 => /usr/lib/libcurl.so.4 (0x00007fb463283000)
        libsmime3.so => /usr/lib/libsmime3.so (0x00007fb463259000)
        libnss3.so => /usr/lib/libnss3.so (0x00007fb463129000)
        libnspr4.so => /usr/lib/libnspr4.so (0x00007fb4630e7000)
        libopenjp2.so.7 => /usr/lib/libopenjp2.so.7 (0x00007fb46308a000)
        liblcms2.so.2 => /usr/lib/liblcms2.so.2 (0x00007fb462e2f000)
        libtiff.so.5 => /usr/lib/libtiff.so.5 (0x00007fb462ba7000)
        /lib64/ld-linux-x86-64.so.2 => /usr/lib64/ld-linux-x86-64.so.2 (0x00007fb46a2bf000)
        libdbus-1.so.3 => /usr/lib/libdbus-1.so.3 (0x00007fb462b52000)
        libatspi.so.0 => /usr/lib/libatspi.so.0 (0x00007fb462b1e000)
        libgraphite2.so.3 => /usr/lib/libgraphite2.so.3 (0x00007fb462af9000)
        libexpat.so.1 => /usr/lib/libexpat.so.1 (0x00007fb462abd000)
        libuuid.so.1 => /usr/lib/libuuid.so.1 (0x00007fb462ab4000)
        libdatrie.so.1 => /usr/lib/libdatrie.so.1 (0x00007fb4628ab000)
        libXau.so.6 => /usr/lib/libXau.so.6 (0x00007fb4626a7000)
        libXdmcp.so.6 => /usr/lib/libXdmcp.so.6 (0x00007fb4624a1000)
        libblkid.so.1 => /usr/lib/libblkid.so.1 (0x00007fb46244c000)
        libnghttp2.so.14 => /usr/lib/libnghttp2.so.14 (0x00007fb462424000)
        libidn2.so.0 => /usr/lib/libidn2.so.0 (0x00007fb462205000)
        libssh2.so.1 => /usr/lib/libssh2.so.1 (0x00007fb461fd7000)
        libpsl.so.5 => /usr/lib/libpsl.so.5 (0x00007fb461dc7000)
        libssl.so.1.1 => /usr/lib/libssl.so.1.1 (0x00007fb461d37000)
        libcrypto.so.1.1 => /usr/lib/libcrypto.so.1.1 (0x00007fb461a65000)
        libgssapi_krb5.so.2 => /usr/lib/libgssapi_krb5.so.2 (0x00007fb461817000)
        libkrb5.so.3 => /usr/lib/libkrb5.so.3 (0x00007fb46152c000)
        libk5crypto.so.3 => /usr/lib/libk5crypto.so.3 (0x00007fb4612f9000)
        libcom_err.so.2 => /usr/lib/libcom_err.so.2 (0x00007fb4612f3000)
        libnssutil3.so => /usr/lib/libnssutil3.so (0x00007fb4612c0000)
        libplds4.so => /usr/lib/libplds4.so (0x00007fb4612bb000)
        libplc4.so => /usr/lib/libplc4.so (0x00007fb4612b4000)
        libsystemd.so.0 => /usr/lib/libsystemd.so.0 (0x00007fb461226000)
        libunistring.so.2 => /usr/lib/libunistring.so.2 (0x00007fb460ea6000)
        libkrb5support.so.0 => /usr/lib/libkrb5support.so.0 (0x00007fb460c99000)
        libkeyutils.so.1 => /usr/lib/libkeyutils.so.1 (0x00007fb460c93000)
        liblz4.so.1 => /usr/lib/liblz4.so.1 (0x00007fb460c72000)
        libgcrypt.so.20 => /usr/lib/libgcrypt.so.20 (0x00007fb460b53000)
        libgpg-error.so.0 => /usr/lib/libgpg-error.so.0 (0x00007fb460933000)
````
</details> 

<details> 
<summary>lddtree output</summary>
xournalpp => ./xournalpp (interpreter => /lib64/ld-linux-x86-64.so.2)
    libboost_system.so.1.68.0 => /usr/lib/libboost_system.so.1.68.0
        librt.so.1 => /usr/lib/librt.so.1
    libboost_filesystem.so.1.68.0 => /usr/lib/libboost_filesystem.so.1.68.0
    libboost_locale.so.1.68.0 => /usr/lib/libboost_locale.so.1.68.0
        libicudata.so.63 => /usr/lib/libicudata.so.63
        libicui18n.so.63 => /usr/lib/libicui18n.so.63
        libicuuc.so.63 => /usr/lib/libicuuc.so.63
            libdl.so.2 => /usr/lib/libdl.so.2
    libboost_iostreams.so.1.68.0 => /usr/lib/libboost_iostreams.so.1.68.0
        libbz2.so.1.0 => /usr/lib/libbz2.so.1.0
        liblzma.so.5 => /usr/lib/liblzma.so.5
    libboost_thread.so.1.68.0 => /usr/lib/libboost_thread.so.1.68.0
    libboost_regex.so.1.68.0 => /usr/lib/libboost_regex.so.1.68.0
    libboost_chrono.so.1.68.0 => /usr/lib/libboost_chrono.so.1.68.0
    libboost_date_time.so.1.68.0 => /usr/lib/libboost_date_time.so.1.68.0
    libboost_atomic.so.1.68.0 => /usr/lib/libboost_atomic.so.1.68.0
    libgtk-3.so.0 => /usr/lib/libgtk-3.so.0
        libX11.so.6 => /usr/lib/libX11.so.6
            libxcb.so.1 => /usr/lib/libxcb.so.1
                libXau.so.6 => /usr/lib/libXau.so.6
                libXdmcp.so.6 => /usr/lib/libXdmcp.so.6
        libXi.so.6 => /usr/lib/libXi.so.6
            libXext.so.6 => /usr/lib/libXext.so.6
        libXfixes.so.3 => /usr/lib/libXfixes.so.3
        libatk-bridge-2.0.so.0 => /usr/lib/libatk-bridge-2.0.so.0
            libdbus-1.so.3 => /usr/lib/libdbus-1.so.3
                libsystemd.so.0 => /usr/lib/libsystemd.so.0
                    liblz4.so.1 => /usr/lib/liblz4.so.1
                    libgcrypt.so.20 => /usr/lib/libgcrypt.so.20
                        libgpg-error.so.0 => /usr/lib/libgpg-error.so.0
            libatspi.so.0 => /usr/lib/libatspi.so.0
        libepoxy.so.0 => /usr/lib/libepoxy.so.0
        libharfbuzz.so.0 => /usr/lib/libharfbuzz.so.0
            libgraphite2.so.3 => /usr/lib/libgraphite2.so.3
        libfontconfig.so.1 => /usr/lib/libfontconfig.so.1
            libexpat.so.1 => /usr/lib/libexpat.so.1
            libuuid.so.1 => /usr/lib/libuuid.so.1
        libfreetype.so.6 => /usr/lib/libfreetype.so.6
            libpng16.so.16 => /usr/lib/libpng16.so.16
    libgdk-3.so.0 => /usr/lib/libgdk-3.so.0
        libXinerama.so.1 => /usr/lib/libXinerama.so.1
        libXrandr.so.2 => /usr/lib/libXrandr.so.2
            libXrender.so.1 => /usr/lib/libXrender.so.1
        libXcursor.so.1 => /usr/lib/libXcursor.so.1
        libXcomposite.so.1 => /usr/lib/libXcomposite.so.1
        libXdamage.so.1 => /usr/lib/libXdamage.so.1
        libxkbcommon.so.0 => /usr/lib/libxkbcommon.so.0
        libwayland-cursor.so.0 => /usr/lib/libwayland-cursor.so.0
        libwayland-egl.so.1 => /usr/lib/libwayland-egl.so.1
        libwayland-client.so.0 => /usr/lib/libwayland-client.so.0
            libffi.so.6 => /usr/lib/libffi.so.6
    libpangocairo-1.0.so.0 => /usr/lib/libpangocairo-1.0.so.0
    libpango-1.0.so.0 => /usr/lib/libpango-1.0.so.0
        libthai.so.0 => /usr/lib/libthai.so.0
            libdatrie.so.1 => /usr/lib/libdatrie.so.1
        libfribidi.so.0 => /usr/lib/libfribidi.so.0
    libatk-1.0.so.0 => /usr/lib/libatk-1.0.so.0
    libcairo-gobject.so.2 => /usr/lib/libcairo-gobject.so.2
    libcairo.so.2 => /usr/lib/libcairo.so.2
        libpixman-1.so.0 => /usr/lib/libpixman-1.so.0
        libxcb-shm.so.0 => /usr/lib/libxcb-shm.so.0
        libxcb-render.so.0 => /usr/lib/libxcb-render.so.0
    libgdk_pixbuf-2.0.so.0 => /usr/lib/libgdk_pixbuf-2.0.so.0
    libgio-2.0.so.0 => /usr/lib/libgio-2.0.so.0
        libmount.so.1 => /usr/lib/libmount.so.1
            libblkid.so.1 => /usr/lib/libblkid.so.1
        libresolv.so.2 => /usr/lib/libresolv.so.2
    libgobject-2.0.so.0 => /usr/lib/libgobject-2.0.so.0
    libglib-2.0.so.0 => /usr/lib/libglib-2.0.so.0
        libpcre.so.1 => /usr/lib/libpcre.so.1
    libgthread-2.0.so.0 => /usr/lib/libgthread-2.0.so.0
    libxml2.so.2 => /usr/lib/libxml2.so.2
    libgmodule-2.0.so.0 => /usr/lib/libgmodule-2.0.so.0
    libpangoft2-1.0.so.0 => /usr/lib/libpangoft2-1.0.so.0
    libpangoxft-1.0.so.0 => /usr/lib/libpangoxft-1.0.so.0
        libXft.so.2 => /usr/lib/libXft.so.2
    libgdk-x11-2.0.so.0 => /usr/lib/libgdk-x11-2.0.so.0
    libgtk-x11-2.0.so.0 => /usr/lib/libgtk-x11-2.0.so.0
    libpoppler.so.78 => /usr/lib/libpoppler.so.78
        libjpeg.so.8 => /usr/lib/libjpeg.so.8
        libcurl.so.4 => /usr/lib/libcurl.so.4
            libnghttp2.so.14 => /usr/lib/libnghttp2.so.14
            libidn2.so.0 => /usr/lib/libidn2.so.0
                libunistring.so.2 => /usr/lib/libunistring.so.2
            libssh2.so.1 => /usr/lib/libssh2.so.1
            libpsl.so.5 => /usr/lib/libpsl.so.5
            libssl.so.1.1 => /usr/lib/libssl.so.1.1
            libcrypto.so.1.1 => /usr/lib/libcrypto.so.1.1
            libgssapi_krb5.so.2 => /usr/lib/libgssapi_krb5.so.2
                libkrb5support.so.0 => /usr/lib/libkrb5support.so.0
                libkeyutils.so.1 => /usr/lib/libkeyutils.so.1
            libkrb5.so.3 => /usr/lib/libkrb5.so.3
            libk5crypto.so.3 => /usr/lib/libk5crypto.so.3
            libcom_err.so.2 => /usr/lib/libcom_err.so.2
        libsmime3.so => /usr/lib/libsmime3.so
            libnssutil3.so => /usr/lib/libnssutil3.so
            libplds4.so => /usr/lib/libplds4.so
            libplc4.so => /usr/lib/libplc4.so
        libnss3.so => /usr/lib/libnss3.so
        libnspr4.so => /usr/lib/libnspr4.so
        libopenjp2.so.7 => /usr/lib/libopenjp2.so.7
        liblcms2.so.2 => /usr/lib/liblcms2.so.2
        libtiff.so.5 => /usr/lib/libtiff.so.5
    libpoppler-glib.so.8 => /usr/lib/libpoppler-glib.so.8
    libz.so.1 => /usr/lib/libz.so.1
    libpthread.so.0 => /usr/lib/libpthread.so.0
    libstdc++.so.6 => /usr/lib/libstdc++.so.6
    libm.so.6 => /usr/lib/libm.so.6
    libgcc_s.so.1 => /usr/lib/libgcc_s.so.1
    libc.so.6 => /usr/lib/libc.so.6

</details>

<details>
<summary>Cmake output on standard CMakeLists.txt</summary>

$ cmake ..
-- The CXX compiler identification is GNU 8.2.1
-- The C compiler identification is GNU 8.2.1
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Found PkgConfig: /usr/bin/pkg-config (found version "1.5.4") 
-- Performing Test GNUXX11
-- Performing Test GNUXX11 - Success
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Looking for pthread_create
-- Looking for pthread_create - not found
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - found
-- Found Threads: TRUE  
-- Boost version: 1.68.0
-- Found the following Boost libraries:
--   system
--   filesystem
--   locale
--   iostreams
--   thread
--   regex
--   chrono
--   date_time
--   atomic
-- Checking for module 'gtk+-3.0 >= 3.18.9'
--   Found gtk+-3.0 , version 3.24.1
-- Checking for module 'glib-2.0 >= 2.32.0'
--   Found glib-2.0 , version 2.58.1
-- Checking for module 'gthread-2.0 >= 2.4.0'
--   Found gthread-2.0 , version 2.58.1
-- Checking for module 'libxml-2.0 >= 2.0.0'
--   Found libxml-2.0 , version 2.9.8
-- Checking for module 'glib-2.0'
--   Found glib-2.0, version 2.58.1
-- Found GTK2_GTK: /usr/lib/libgtk-x11-2.0.so  
-- Checking for modules 'poppler;poppler-glib'
--   Found poppler, version 0.67.0
--   Found poppler-glib, version 0.67.0
-- Found ZLIB: /usr/lib/libz.so (found version "1.2.11") 
-- Found Gettext: /usr/bin/xgettext (found version "0.19.8.1") 
Poppler sources exist
-- Looking for include file fcntl.h
-- Looking for include file fcntl.h - found
-- Looking for include file stdlib.h
-- Looking for include file stdlib.h - found
-- Looking for include file string.h
-- Looking for include file string.h - found
-- Looking for include file unistd.h
-- Looking for include file unistd.h - found

Configuration:
        Compiler:                   /usr/bin/c++
        Mathtex enabled:            ON
        CppUnit enabled:            OFF

Unstable features:
        Layers sidebar:             OFF

Static libraries:
        Poppler:                    OFF
        Boost:                      OFF

-- Configuring done
-- Generating done
-- Build files have been written to: /home/siliconninja/Documents/Linux Compiled Programs/xournalpp_gtkworkingtest/xournalpp/build


</details>

Finally, I tried reducing the amount of dependencies in the CMakeLists.txt file by commenting some of it out. I wanted to get rid of the ``-- Found GTK2_GTK: /usr/lib/libgtk-x11-2.0.so`` line when running ``cmake ..`` in the `build` directory because that had nothing to do with GTK3. Nothing was listed under gtk2_gtk line (usually libraries that depend on GTK2 and messages) but there were plenty of libraries listed under the GTK3 line. I commented out so many lines before I finally got that to happen.

When I got the new file from GitHub (since there were so many commits today) I tried commenting out just the lines to stop that GTK2_GTK line from showing up. I couldn't isolate them. I'm taking a break from this right now so I'll just post what CMakeLists.txt file I got to work (it's from an earlier revision today).
[CMakeLists-NOGTK2_GTK_line.txt](https://github.com/xournalpp/xournalpp/files/2606239/CMakeLists-NOGTK2_GTK_line.txt)

I didn't compile it because I didn't expect it to work. Hopefully that saves time for everyone else trying to work on this issue.

This problem was a lot harder than I thought. It wasn't as simple as a conflicting library. Probably one of the CMakeLists.txt files that depends on GTK2 inside a folder or something...

**Comment by siliconninja** [comment]
@andreasb242 You're welcome!
I've tried again. I deleted all the source files this time, except for the stuff in the cmake directory, because that's required for building. Attached is my CMakeLists.txt.
[CMakeLists.txt](https://github.com/xournalpp/xournalpp/files/2608947/CMakeLists.txt)

The contents of my setup are:
````
$ tree -L 2
.
├── ABOUT-NLS
├── AUTHORS
├── build
│   ├── cmake
│   ├── CMakeCache.txt
│   ├── CMakeFiles
│   ├── cmake_install.cmake
│   ├── example-0
│   └── Makefile
├── cmake
│   ├── cmake_uninstall.cmake.in
│   ├── find
│   ├── include
│   └── README.md
├── CMakeLists.txt
├── doc
│   └── ui
├── example-0.c
├── LICENSE
├── readme
│   └── main.png
└── README.md

9 directories, 13 files
````

I was able to get the example-0.c file to compile properly using cmake and make.

I had to include a few more lines to get GTK3 to work on that example (my error before was
`fatal error: gtk/gtk.h: No such file or directory #include <gtk/gtk.h>`) from this link: https://gist.github.com/fracek/3323924
For some reason it ran with no issues.

![screenshot_20181122_121938](https://user-images.githubusercontent.com/29907828/48916587-ea453a80-ee50-11e8-95d7-abe0a201308f.png)

I wonder if we have to do the same thing (add those lines) for the (build/)src/xournalpp binary that's being created (since it's being created in the src/ folder). Maybe we're not linking stuff properly, maybe the GTK2 stuff is being linked instead of the GTK3 stuff, or both are (most likely).

https://github.com/xournalpp/xournalpp/blob/master/src/CMakeLists.txt#L84
https://github.com/xournalpp/xournalpp/blob/master/src/CMakeLists.txt#L94

**Comment by andreasb242** [comment]
Example
>PKG_CHECK_MODULES(GTK3 REQUIRED gtk+-3.0)
>INCLUDE_DIRECTORIES(${GTK3_INCLUDE_DIRS})

here it's called GTK3, because of this all vars starting with GTK3

Xournal++
>pkg_check_modules (GTK REQUIRED "gtk+-3.0 >= 3.18.9")
>add_includes_ldflags ("${GTK_LDFLAGS}" "${GTK_INCLUDE_DIRS}")

Here it's called GTK, therefore all vars start with GTK.

But maybe a simple

INCLUDE_DIRECTORIES(${GTK_INCLUDE_DIRS})
Would solve it.

I don't know the difference from your build commands.
i'm really not a CMake Specialist... (else I may already fixed the problem...)

**Comment by andreasb242** [comment]
gtk-x11-2.0 is probably the problem.

With
>cmake --graphviz=test.dot
we can output the dependencies.
>dot -Tps test.dot -o outfile.ps

It shows, that xournalpp directly references gtk-x11-2.0 !?

----

I found it!

**Comment by siliconninja** [comment]
Wow! Yeah I found gtk-x11-2.0 in lddtree but I wasn't sure where it was coming from. That's why `-- Found GTK2_GTK: /usr/lib/libgtk-x11-2.0.so` was being printed when I ran cmake and both GTK2 and GTK3 libraries were being included. When I was debugging, I tried `grep -nr "gtk" .` but that only gave a bunch of results of "gtk" in source files. I was also trying "gtk2" but grep didn't find "GTK2". I should have redirected it to a file instead so I could see the full output. Well I learned something from this experience!

The command I should have used:
````
$ grep -nri "GTK" . > output2.txt
grep: input file ‘./output.txt’ is also the output
# or...

$ grep -nri "GTK2" . > output2.txt
grep: input file ‘./output2.txt’ is also the output

$ cat output2.txt | head -n 50
./CMakeLists.txt:282:option (DEV_CHECK_GTK3_COMPAT "Adds a few compiler flags to check basic GTK3 upgradeability support (still compiles for GTK2!)")
./README.md:61:dnf install cmake libglade2-devel texlive-scheme-basic texlive-dvipng glibmm24-devel gtk2-devel gtk+-devel boost boost-devel poppler-glib-devel
./cmake/find/FindPoppler.cmake:17:    find_path(GTK2_GLIBCONFIG_INCLUDE_DIR
./cmake/find/FindPoppler.cmake:23:find_package(GTK2)
./cmake/find/FindPoppler.cmake:79:set(POPPLER_INCLUDE_DIRS ${GTK2_INCLUDE_DIRS} ${POPPLER_INCLUDE_DIR} ${POPPLER_GLIB_INCLUDE_DIR})
./cmake/find/FindPoppler.cmake:80:set(POPPLER_LIBRARIES ${GTK2_LIBRARIES} ${POPPLER_LIBRARY} ${POPPLER_GLIB_LIBRARY})
./cmake/README.md:41:| `DEV_CHECK_GTK3_COMPAT` *[A]*  | OFF              | Adds a few compiler flags to check basic GTK3 upgradeability support (still compiles for GTK2!)
./.git/packed-refs:3:bb02242af3dc1ddd2a771b523dded0c5ec7f0dc3 refs/remotes/origin/master_gtk2_stable
Binary file ./.git/objects/pack/pack-d42e3e352cd044e5157f017260d3179e9a9a8791.pack matches
./src/pdf/popplerdirect/workaround/poppler-0.67.0/ChangeLog:41846:    [CMake] switch gtk2 (gtk+, gdk-pixbuf, gthread, gio) search to use
./src/util/GtkColorWrapper.h:4: * Helper functions to wrap GdkColor/GdkRGBA color types used by GTK2 / GTK3

````

Hiding in plain sight...

Why are things like `GTK2_GLIBCONFIG_INCLUDE_DIR` and `${GTK2_LIBRARIES}` still there though? Shouldn't we be using GTK3 now? Does Poppler require GTK2 libraries to compile? I'll experiment with that file and create a separate PR that doesn't include GTK2 stuff.

**Comment by andreasb242** [comment]
I merged the pull now - so the main Problem is solved (and the ticket closed)

@siliconninja i saw this also, but after it was working I wouldn't change anything ;-)
Feel free to create a Pull request which cleans this also - I'll merge it.

**Comment by siliconninja** [comment]
Thank you! I've created a pull request #360 with those changes and fixed a typo in CMakeLists.txt that I found.
